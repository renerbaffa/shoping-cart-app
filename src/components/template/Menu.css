.Menu-container {
  color: lightgray;
  display: flex;
  justify-content: center;
}

.Menu-height {
  height: 36px;
  border: 1px solid #434343;
}
