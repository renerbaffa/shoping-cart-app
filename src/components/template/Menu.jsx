import React from 'react';
import cx from 'classnames';

import MenuItem from './MenuItem';

import './Menu.css';

const Menu = () => (
  <div className="Menu-container">
    <div className={cx('limited-width', 'gray-background')}>
      <MenuItem label="Home" />
      <MenuItem label="Top-5 Products" />
      <MenuItem label="Your Cart" />
    </div>
  </div>
);

export default Menu;
