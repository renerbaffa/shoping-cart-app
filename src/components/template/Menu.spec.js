import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import CART from '../../mocks/Cart';

import Menu from './Menu';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<Menu />', () => {
  it('should show spinner correctly', () => {
    const store = mockStore({
      cart: CART,
    });
    const menu = renderer.create(
      <Provider store={store}>
        <Router>
          <Menu />
        </Router>
      </Provider>
    ).toJSON();
    expect(menu).toMatchSnapshot();
  });
})