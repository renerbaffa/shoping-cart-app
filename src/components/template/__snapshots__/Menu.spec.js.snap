// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should show spinner correctly 1`] = `
<div
  className="Menu-container"
>
  <div
    className="limited-width gray-background space-between Menu-height"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "lightgray",
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MenuItem-container"
      >
        Home
      </div>
    </a>
    <div
      className="MenuItem-container"
    >
      Top-5 Products
    </div>
    <a
      href="my-cart"
      onClick={[Function]}
      style={
        Object {
          "color": "lightgray",
          "textDecoration": "none",
        }
      }
    >
      <div>
        <div
          className="MenuItem-container Menu-cart"
        >
          Your Cart
        </div>
        <div
          className="ProductBadge-container"
        >
          5
        </div>
      </div>
    </a>
  </div>
</div>
`;
